### ISSUE TITLE
Filtering of DataFrame based on the column with dates in Date32 format is not working [2]

### ISSUE THREAD

**Issue by solisortusaeternum** [pola-rs/polars#1342]
#### Are you using Python or Rust?

**Python**

#### What version of polars are you using?

polars==0.9.5

#### What operating system are you using polars on?

Ubuntu 20.04.1 LTS

#### Describe your bug.

Polars DataFrame is returning wrong results when comparison operation is performed on dates using pl.lit_date().

#### What are the steps to reproduce the behavior?

```
import polars as pl
from datetime import datetime

# create a simple dataset
dataset = pl.DataFrame({"date": ["2020-01-02", "2020-01-03", "2020-01-04"], "index": [1, 2, 3]})
df = dataset.with_column(pl.col("date").str.strptime(pl.datatypes.Date32, "%Y-%m-%d"))

# filter out the data to match only records from the previous year
df.filter(col('date') <= pl.lit_date(datetime(2020, 1, 5)))
```

#### What is the actual behavior?

```
df.filter(col('date') <= pl.lit_date(datetime(2020, 1, 5)))

shape: (0, 2)
┌──────────────┬───────┐
│ date         ┆ index │
│ ---          ┆ ---   │
│ date32(days) ┆ i64   │
╞══════════════╪═══════╡
└──────────────┴───────┘
```

#### What is the expected behavior?

The same DataFrame with 3 row and 3 columns should be returned as the result.

**Comment by ritchie46** [comment]
Could you try version `0.9.5`?

**Comment by solisortusaeternum** [comment]
> Could you try version `0.9.5`?

@ritchie46 yeah, I tried it with 0.9.5, I put 0.9.4 by mistake in the issue description.

**Comment by ritchie46** [comment]
Check. I've got the issue as well. Can you  use `pl.lit()`. That should work.

**Comment by ritchie46** [comment]
@solisortusaeternum this was fixed, but I forgot to cherry-pick the commit that fixed it. It is resolved in #1344.

However, `pl.lit()` does not have this issue. You can use that.

I also deprecated `lit_date` as its a bit redundant.

**Comment by solisortusaeternum** [comment]
> @solisortusaeternum this was fixed, but I forgot to cherry-pick the commit that fixed it. It is resolved in #1344.
> 
> However, `pl.lit()` does not have this issue. You can use that.
> 
> I also deprecated `lit_date` as its a bit redundant.

thanks a lot, `pl.lit()` worked.
